packages/blade/src/components/Amount/Amount.stories.tsx

```diff
@@ -217,3 +223,98 @@ StrikeThrough.args = {
   isStrikethrough: true,
 };
 StrikeThrough.storyName = 'Strike Through';
+
+// TODO: Replace below with i18nify getDefaultLocales API
+const localeList = [
+  {
+    country: 'India',
+    locale: 'en-IN',
+  },
+  {
+    country: 'USA',
+    locale: 'en-US',
+  },
+  {
+    country: 'Malaysia',
+    locale: 'ms-MY',
+  },
+  {
+    country: 'France',
+    locale: 'fr-FR',
+  },
+  {
+    country: 'Germany',
+    locale: 'de-DE',
+  },
+];
+
+const I18nAmountWrapper = (args: AmountProps): JSX.Element => {
+  const { setI18nState } = useI18nContext();
+  const [currency, setCurrency] = useState('INR');
+
+  return (
+    <>
+      <AmountComponent {...args} currency={currency as AmountProps['currency']} />
+      <Divider marginY="spacing.4" marginTop="spacing.8" />
+      <Dropdown selectionType="single">
+        <SelectInput label="Select currency" />
+        <DropdownOverlay>
+          <ActionList>
+            {Object.keys(getCurrencyList()).map((value) => (
+              <ActionListItem
+                key={value}
+                title={value}
+                value={value}
+                onClick={({ name }) => {
+                  setCurrency(name);
+                }}
+              />
+            ))}
+          </ActionList>
+        </DropdownOverlay>
+      </Dropdown>
+      <Divider marginY="spacing.4" />
+      <Dropdown selectionType="single">
+        <SelectInput label="Select locale" />
+        <DropdownOverlay>
+          <ActionList>
+            {localeList.map((item) => (
+              <ActionListItem
+                key={item.locale}
+                title={`${item.country}(${item.locale})`}
+                value={item.locale}
+                onClick={({ name }) => {
+                  setI18nState?.({ locale: name });
```

Comment:
@tarun-khanna this is setting the context at a global amount's level ideally we shall scope it within this story itself?

Why?
I opened the amount story page i.e default `Docs` page and saw the currency in some default locale then I navigated to the locale story and was playing around with it and then went back to the Docs page of amount and it was pointing to the locale I set under the locale story which was misleading as there's no information mentioned about that

Reply:
Setting a state by I18nProvider's `setI18nState` will always set it globally for `i18nify-js`. We decided to architect our library in this fashion so to deal with out of sync state issues between sidebar, header, body and other elements as discussed in our earlier meetings. 

To deal with this, 
1. We can introduce an internal `locale` prop hidden to consumers.
2. Then on locale dropdown click, We don't have to call `setI18nState` but can maintain an internal locale state which will be passed to the Amount instance used in this story.
3. We can then pass `locale` prop to i18nify-js `formatNumber` api which accepts this locale param for overriding global state. 

Thoughts ? @kamleshchandnani

Reply:
Okay i'm saying keep it simple from demonstration point of view because right now there's no co-relation to the end consumer that changes in one story is affecting other. what I'm suggesting is wrap another i18n context around just the locale story so the state changes is only affected to that story and not goes all the stories of amount component.

Reply:
As discussed earlier, there is no finalised approach here that will resolve this issue so we will be addressing this issue in the next PR. The current one can still go as it is. 
Attaching MOM [here](https://razorpay.slack.com/archives/C05RBSKC6KD/p1707485708148249?thread_ts=1705482465.156459&cid=C05RBSKC6KD)

cc @kamleshchandnani